apps/server/lib/lexical/server/provider/queue.ex

```diff
@@ -44,7 +44,7 @@ defmodule Lexical.Server.Provider.Queue do
     @spec cancel(t, pos_integer()) :: t
     def cancel(%__MODULE__{} = state, request_id) do
       with {:ok, %Task{} = task} <- Map.fetch(state.tasks_by_id, request_id),
-           true <- Process.exit(task.pid, :cancelled) do
+           true <- Process.exit(task.pid, :normal) do
```

Comment:
Why change this?

this change will cause a completions bug in neovim.

<img width="1072" alt="image" src="https://github.com/lexical-lsp/lexical/assets/12830256/704c933d-0657-4f42-85dd-aa08d784177d">

just try to create a function by typing: `def foo|` can replicate this error.

Reply:
```lua

[ERROR][2023-06-13 08:35:21] .../vim/lsp/rpc.lua:493	"No callback found for server response id 3"
[ERROR][2023-06-13 08:35:21] .../lua/vim/lsp.lua:1086	"LSP[lexical]"	"on_error"	{
  code = "NO_RESULT_CALLBACK_FOUND",
  err = {
    id = 3,
    jsonrpc = "2.0",
    result = { {
        detail = "",
        insertText = "def ${1:name}($2) do\n  $0\nend\n",
        insertTextFormat = 2,
        kind = 7,
        label = "def (Define a function)",
        sortText = " def (Define a function)"
      }, {
        detail = "",
        insertText = "defdelegate ${1:call}, to: ${2:module} $0\n",
        insertTextFormat = 2,
        kind = 7,
        label = "defdelegate (Define a delegate function)",
        sortText = "%defdelegate (Define a delegate function)"
      }, {
        detail = "",
        insertText = "defexception [${1:fields}] $0\n",
        insertTextFormat = 2,
        kind = 7,
        label = "defexception (Define an exception)",
        sortText = "%defexception (Define an excepti
        ```

Reply:
not in an editor that _isn't_ neovim; What is going on here?

Reply:
> Why change this?

Because it was logging an error because of a totally normal thing; the task was canceled by the client. No big deal.

> this change will cause a completions bug in neovim.

How? This should be completely internal to the VM, the task still gets canceled by the client.

Reply:
Let me try cancelling this a different way, and see if it helps.

Reply:
I've pushed a change, see if it works for you.